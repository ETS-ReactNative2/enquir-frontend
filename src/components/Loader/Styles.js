import { StyleSheet } from 'react-native';

const containers = StyleSheet.create({
    modalBackground: {
        flex: 1,
        alignItems: 'center',
        flexDirection: 'column',
        justifyContent: 'space-around',
        backgroundColor: '#00000040'
      },
      activityIndicatorWrapper: {
        backgroundColor: '#FFFFFF',
        height: 120,
        width: 130,
        borderRadius: 10,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
      },
});

const text = StyleSheet.create({
    loadText: {
        alignSelf: 'center',
        marginBottom: 20,
    },
});

export { containers, text };