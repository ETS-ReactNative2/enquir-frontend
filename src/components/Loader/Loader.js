import React from 'react';
import {
  View,
  Modal,
  ActivityIndicator,
  Text,
} from 'react-native';
import { containers, text } from './Styles';

// Credit to Kelley Rose (src: https://medium.com/@kelleyannerose/react-native-activityindicator-for-a-quick-easy-loading-animation-593c06c044dc)

const Loader = props => {
  const {
    loading,
    loadText,
    ...attributes
  } = props;
  return (
    <Modal
      transparent={true}
      animationType={'none'}
      visible={loading}>
      <View style={containers.modalBackground}>
        <View style={containers.activityIndicatorWrapper}>
          <Text style={text.load}>{loadText}</Text>
          <ActivityIndicator
            animating={loading} />
        </View>
      </View>
    </Modal>
  )
};

export default Loader;