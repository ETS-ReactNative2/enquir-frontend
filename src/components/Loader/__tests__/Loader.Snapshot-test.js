import React from 'react';
import renderer from 'react-test-renderer';

import Loader from '../Loader';

test('renders a Loader using Snapshots and is not loading', () => {
    expect(renderer.create(
        <Loader
            loading={false}
            loadText={''}
        />
    )).toMatchSnapshot();
});

test('renders a Loader using Snapshots and is loading', () => {
    expect(renderer.create(
        <Loader
            loading={true}
            loadText={''}
        />
    )).toMatchSnapshot();
});