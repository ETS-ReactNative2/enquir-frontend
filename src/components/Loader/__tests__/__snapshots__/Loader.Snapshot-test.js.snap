// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Loader using Snapshots and is loading 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000040",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "display": "flex",
          "height": 120,
          "justifyContent": "center",
          "width": 130,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontFamily": "Roboto-Light",
            "fontSize": 16,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</Modal>
`;

exports[`renders a Loader using Snapshots and is not loading 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000040",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "display": "flex",
          "height": 120,
          "justifyContent": "center",
          "width": 130,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontFamily": "Roboto-Light",
            "fontSize": 16,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
      <ActivityIndicator
        animating={false}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</Modal>
`;
